trieste/acquisition/function/function.py

```diff
@@ -583,6 +583,217 @@ def _update_expected_improvement_fn(
             self._expected_improvement_fn.update(eta)  # type: ignore
 
 
+class MonteCarloExpectedImprovement(SingleModelAcquisitionBuilder[ProbabilisticModel]):
+    """
+    Builder for a Monte Carlo-based expected improvement function for use with a model without
+    analytical expected improvement (e.g. a deep GP). The "best" value is taken to be
+    the minimum of the posterior mean at observed points.
+    """
+
+    def __init__(self, sample_size: int):
+        """
+        :param sample_size: The number of samples for each batch of points.
+        :raise tf.errors.InvalidArgumentError: If ``sample_size`` is not positive.
+        """
+        tf.debugging.assert_positive(sample_size)
+
+        super().__init__()
+
+        self._sample_size = sample_size
+
+    def __repr__(self) -> str:
+        """"""
+        return f"MonteCarloExpectedImprovement({self._sample_size!r})"
+
+    def prepare_acquisition_function(
+        self,
+        model: ProbabilisticModel,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param model: The model over the specified ``dataset``. Must have output dimension [1].
+        :param dataset: The data from the observer. Cannot be empty.
+        :return: The estimated *expected improvement* acquisition function.
+        :raise ValueError (or InvalidArgumentError): If ``dataset`` is not populated, ``model``
+            does not have an output dimension of [1] or does not have a ``reparam_sample`` method.
+        """
+        get_sampler = getattr(model, "reparam_sampler", None)
+        if not callable(get_sampler):
+            raise ValueError(
+                "The model must have a `reparam_sampler` method, which takes `num_samples` and "
+                "which returns a reparametrization `ReparametrizationSampler` to create smooth "
+                "samples at query points."
+            )
+
+        sampler = get_sampler(self._sample_size)
```

Comment:
since the sampler accepts the jitter argument I think we should add it here so that user could specify it when setting up the acq function - _init__ and repr will need to be adjusted